core/src/main/java/org/radargun/stats/Request.java

```diff
@@ -21,9 +21,16 @@
    private long responseCompleteTime = Long.MIN_VALUE;
    private boolean successful = true;
 
+   private final long intendedTime;
```

Comment:
So I still don't see why do you need to store `intendedTime` as a separate field, and then have a special handling in HistogramOperationStats. I would just let stage provide this 'fake' time and keep statistics as-is.